### PR TITLE
Fix publishing api republishing

### PR DESCRIPTION
**Pull request by jamiecobbett**
Most of the interesting changes are in the last commit: 

Until now, it was easy to accidentally republish Editions in a way that could
publish the wrong content or lose the draft version from publishing-api.

This is because the implementation was originally written when the draft and
live content items were separate resources. That meant that you could
independently push live and draft items without them affecting each other.

Now, however, content is published by promoting it from draft to live, via the
publishing-api's publish action.

This means that the order of requests, when republishing the draft and live
editions of the same document, matters, because otherwise you could publish the
draft edition or overwrite the draft edition with the live edition.

I note that I have broken lots of calls to DataHygiene::PublishingApiRepublisher
that exist under db/data_migrations. There were already some examples
referencing PublishingApiEditionWorker, which hasn't existed for some time.

### CODE REVIEW

File: lib/data_hygiene/publishing_api_document_republisher.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        raise ArgumentError, "The argument to PublishingApiDocumentRepublisher must be a subclass of Edition"
+      end
+      @edition_class = edition_class
+      @logger = logger
+      @queued = 0
+    end
+
+    def perform
+      logger.info "Queuing #{documents.count} #{edition_class} instances for republishing to the Publishing API"
+      documents.each do |document|
```
Comment (benlovell) [inline lib/data_hygiene/publishing_api_document_republisher.rb]: `find_each` may be preferable for batching. For smaller sets it has no discernible overhead.

File: lib/data_hygiene/publishing_api_republisher.rb [hunk 2]
```diff
@@ -1,23 +1,18 @@
...
-  #
-  # Note: Whitehall::PublishingApi will raise an exception if an attempt is made
-  # to publish an instance that is not currently supported (e.g. draft editions
-  # cannot be pubilshed yet). The scope passed in should not include any
-  # instances that are not publishable.
   class PublishingApiRepublisher
-    attr_reader :logger, :scope, :republished
+    attr_reader :logger, :scope, :queued
 
     def initialize(scope, logger=Logger.new(STDOUT))
```
Comment (gpeng) [inline lib/data_hygiene/publishing_api_republisher.rb]: rubocop wants `logger = Logger.new(STDOUT)`

Comment (benlovell) [inline lib/data_hygiene/publishing_api_republisher.rb]: Pshht. Rubocop is just so pedantic. :wink:

**Comment by boffbowsh** [comment]
For expediency and to get tests running, I've pushed the rubocop fix to this branch. Feel free to squash it when you're ready, @jamiecobbett

File: lib/data_hygiene/publishing_api_document_republisher.rb (hunk 1 above)
Comment (jamiecobbett) [inline lib/data_hygiene/publishing_api_document_republisher.rb]: Good spot - rebased.

**Comment by boffbowsh** [comment]
:+1:

**Comment by benilovj** [comment]
I've rebased this.

**Comment by benilovj** [comment]
This appears to work without error on Integration for Case Studies and Take Part pages.

**Pull request merged**